src/source/promises.bs

```diff
@@ -482,7 +488,7 @@ namespace promises.internal
                     print "Crash during utils.delay:", e
                 #end if
             end try
-            m[delayId] = invalid
+            m.delete(delayId)
```

Comment:
I noticed these timer entries were never getting properly cleaned up, as they were only being set to invalid. This should clean up the `m` variable nicely now.